docs/source/technical_tutorials/user_data/memory-profiling-python.ipynb

```diff
@@ -0,0 +1,210 @@
+{
```

Comment:
<p>Can we explain this a little more? Looks like we're adding a decorator to a new function. Can we add the decorator to an existing function or are users going to take all their lines of code and paste it under this function?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/371/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>Thanks, this is a good point. I've added explanation in the updated notebook.</p>

Reply:
<p>I'm still a little confused as to why we have to create a new code_to_profile function. Can we generalize the name of the function and describe that we can add the decorate to our already existing functions? Users might take it that they need to name their function code_to_profile and strip out all of their code from their existing functions</p>

Reply:
<p>Thanks. I agree as to not give any benefit of doubt for confusion. Changes:</p><ul><li>Removed Step 2, no need to add new function</li><li>Generalized function name to my_func()</li><li>Described decorator</li></ul>

Reply:
<p>To Jamison's point, perhaps we can add a bit more clarity here.</p><p> </p><p>How about something like the following for the code example, along with the sentence that follows?</p><p> </p><pre class="ql-syntax" spellcheck="false">@profile
def my_function():
    # Perform some potentially memory-intensive computation
    # ...
</pre><p> </p><p>You may add the <code>@profile</code> decorator to individual functions that you want to profile. This allows you to limit which parts of your program are profiled, thus limiting the volume of profiling output.</p><p> </p><p>However, this requires you to modify your code. If you wish to avoid modifying your code, particularly when it is not yet obvious which parts of your code may be consuming too much memory, simply add the <code>-m memory_profiler</code> to the <code>python</code> command.  </p>